### PR TITLE
fix: file resolution

### PR DESCRIPTION
**Pull request by Jakub-Vacek**
This PR fixes missing dirname in profile, map and provider file resolution and aligns profile caching with: 
https://www.notion.so/superface/Caching-in-SDK-2-0-0-afee3996e7be476982bb4bd599c2a587

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have updated the documentation accordingly. If you are changing **code related to user secrets** you need to really make sure that [security documentation](https://github.com/superfaceai/one-sdk-js/blob/main/SECURITY.md) is correct.
- [x] I have read the [CONTRIBUTING](https://github.com/superfaceai/one-sdk-js/blob/main/CONTRIBUTING.md) document.
- [x] I haven't repeated the code. (DRY)
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/core/profile/resolve-profile-ast.ts [hunk 1]
```diff
@@ -81,53 +81,44 @@ export async function resolveProfileAst({
...
+      // if we find source file we assume compiled file next to it
       if (filepath.endsWith(EXTENSIONS.profile.source)) {
+        astPath = filepath.replace(
+          EXTENSIONS.profile.source,
+          EXTENSIONS.profile.build
+        );
+
+        if (!(await fileSystem.exists(astPath))) {
+          throw sourceFileExtensionFoundError(EXTENSIONS.profile.source);
+        }
```
Comment (freaz) [inline src/core/profile/resolve-profile-ast.ts:103]: issues: if I am reading it right, it checks for presense of ast, but throws error for missing source document. Should it be?

```suggestion
        if (!(await fileSystem.exists(astPath))) {
          throw sourceFileExtensionFoundError(EXTENSIONS.profile.build);
        }
```

**Review by freaz — COMMENTED** [review]
There are two things, that might be an issue.

File: src/core/profile/resolve-profile-ast.ts (hunk 1 above)
Comment (Jakub-Vacek) [reply src/core/profile/resolve-profile-ast.ts:103]: This one is correct - this informs user that .supr (source) file must be compiled.

This case will happen when you are migrating from SDK 1.x to 2.0 with locally linked files

File: src/core/profile/cache-profile-ast.ts [hunk 2]
```diff
@@ -68,41 +69,41 @@ export async function tryToLoadCachedAst({
...
   // Check version
-  if (version !== undefined) {
-    const cachedVersion = `${possibleAst.header.version.major}.${possibleAst.header.version.minor}.${possibleAst.header.version.patch}`;
-    if (possibleAst.header.version.label !== undefined) {
-      version += `-${possibleAst.header.version.label}`;
-    }
+  const cachedVersion = `${possibleAst.header.version.major}.${possibleAst.header.version.minor}.${possibleAst.header.version.patch}`;
+  if (possibleAst.header.version.label !== undefined) {
+    version += `-${possibleAst.header.version.label}`;
+  }
```
[the remark this answers is not in the document]
Comment (Jakub-Vacek) [reply src/core/profile/cache-profile-ast.ts:75]: Yep, I missed that.

File: src/core/profile/resolve-profile-ast.ts (hunk 1 above)
Comment (freaz) [reply src/core/profile/resolve-profile-ast.ts:104]: 🤦 sorry didn't check what is inside `sourceFileExtensionFoundError`. It is alright.

**Review by freaz — APPROVED** [review]
All looks good 👍

**Review by lukas-valenta — APPROVED** [review]

**Pull request merged**